frank/utilities.py

```diff
@@ -458,38 +458,84 @@ def draw_bootstrap_sample(u, v, vis, weights):
     return u_boot, v_boot, vis_boot, weights_boot
 
 
-def sweep_profile(r, I, axis=0):
+def sweep_profile(r, I, project=False, phase_shift=False, geom=None, axis=0,
+                  xmax=None, ymax=None):
     r"""
     Sweep a 1D radial brightness profile over :math:`2 \pi` to yield a 2D
-    brightness distribution
+    brightness distribution. Optionally project this sweep by a supplied
+    geometry.
 
     Parameters
     ----------
     r : array
-          Radial coordinates at which the 1D brightness profile is defined
+        Radial coordinates at which the 1D brightness profile is defined
     I : array
-          Brightness values at r
+        Brightness values at r
+    project : bool, default = False
+        Whether to project the swept profile by the supplied geom
+    phase_shift : bool, default = False
+        Whether to phase shift the projected profile by the supplied geom.
+        If False, the source will be centered in the image
+    geom : SourceGeometry object, default=None
+        Fitted geometry (see frank.geometry.SourceGeometry). Here we use
+        geom.inc [deg], geom.PA [deg], geom.dRA [arcsec], geom.dDec [arcsec] if
+        project=True
     axis : int, default = 0
-          Axis over which to interpolate the 1D profile
+        Axis over which to interpolate the 1D profile
+    xmax, ymax : float, default = None
+        Value setting the x- and y-bounds of the image (same units as r). The
+        positive and negative bounds are both set to this value (modulo sign)
 
     Returns
     -------
     I2D : array, shape = (len(r), len(r))
-        2D brightness distribution
+        2D brightness distribution (projected if project=True)
     xmax : float
         Maximum x-value of the 2D grid
     ymax : float
         Maximum y-value of the 2D grid
 
+    Notes
+    -----
+    Sign convention: a negative geom.dRA shifts the source to the right
+    in the image
     """
+    if project:
+        inc, pa, dra, ddec = geom.inc, geom.PA, geom.dRA, geom.dDec
+        inc *= deg_to_rad
+        pa *= deg_to_rad
+        if not phase_shift:
+            dra, ddec = 0., 0.
+
+        cos_i = np.cos(inc)
+        cos_pa, sin_pa = np.cos(pa), np.sin(pa)
+
+    if not xmax and not ymax:
+        xmax = ymax = r.max()
+    elif not xmax:
+        xmax = r.max()
+    elif not ymax:
+        ymax = r.max()
+    else:
+        pass
```

Comment:
```suggestion
    if not xmax:
        xmax = r.max()
    if not ymax:
        ymax = r.max()
```

Reply:
In my defense, it was a little late ha. fixed in b028bfc